app/views/simulator/edit.html.erb

```diff
@@ -100,93 +106,21 @@
-        </div>
-
-        <div class="panelHeader">Decoders & Plexers</div>
-        <div class="panel">
-          <div class="icon logixModules" id="Multiplexer" ></div>
-          <div class="icon logixModules" id="Demultiplexer" ></div>
-          <div class="icon logixModules" id="BitSelector" ></div>
-          <div class="icon logixModules" id="MSB" ></div>
-          <div class="icon logixModules" id="LSB" ></div>
-          <div class="icon logixModules" id="PriorityEncoder" ></div>
-          <div class="icon logixModules" id="Decoder" ></div>
-        </div>
-
-        <div class="panelHeader">Sequential Elements</div>
-        <div class="panel">
-          <div class="icon logixModules" id="DflipFlop" ></div>
-          <div class="icon logixModules" id="Dlatch" ></div>
-          <div class="icon logixModules" id="TflipFlop" ></div>
-          <div class="icon logixModules" id="JKflipFlop" ></div>
-          <div class="icon logixModules" id="SRflipFlop" ></div>
-          <div class="icon logixModules" id="TTY" ></div>
-          <div class="icon logixModules" id="Keyboard" ></div>
-          <div class="icon logixModules" id="Clock" ></div>
-          <div class="icon logixModules" id="Rom" ></div>
-          <div class="icon logixModules" id="RAM" ></div>
-          <div class="icon logixModules" id="EEPROM" ></div>
-        </div>
-
-        <div class="panelHeader">Test Bench</div>
-        <div class="panel">
-          <div class="icon logixModules" id="TB_Input" ></div>
-          <div class="icon logixModules" id="TB_Output" ></div>
-          <div class="icon logixModules" id="ForceGate" ></div>
-
-        </div>
-        <div class="panelHeader">Misc</div>
-        <div class="panel">
-          <div class="icon logixModules" id="Flag" ></div>
-          <div class="icon logixModules" id="Splitter" ></div>
-          <div class="icon logixModules" id="Adder" ></div>
-          <div class="icon logixModules" id="TriState" ></div>
-          <div class="icon logixModules" id="Buffer" ></div>
-          <div class="icon logixModules" id="ControlledInverter" ></div>
-          <div class="icon logixModules" id="ALU" ></div>
-          <div class="icon logixModules" id="Rectangle" ></div>
-          <div class="icon logixModules" id="Arrow" ></div>
-          <div class="icon logixModules" id="Text" ></div>
-          <div class="icon logixModules" id="Tunnel" ></div>
-        </div>
+        <% CircuitElement.all.order("category").group_by(&:category).each do |category, elements|%>
+          <div class="panelHeader"><%= category %> </div>
+            <div class="panel" style="overflow-y:hidden">
+          <% elements.each do |element| %>
+              <div class="icon logixModules" id="<%= element.name %>" >
+                <img src=<%= element.image.url %>>
+                <p class="img__description"> <%= element.name %></p>
+              </div>
+          <% end %>
+            </div>
+        <% end %>
```

Comment:
This might actually hurt future attempts to decouple the simulator and the backend.